lib/ansible/modules/windows/win_description.ps1

```diff
@@ -0,0 +1,114 @@
+
+$owner = $module.Params.owner
+$organization = $module.Params.organization
+$description = $module.Params.description
+$regPath="HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\"
+
+#Function to create a CIM object
+Function GetCimsObject {
+    param(
+        [Parameter(Position=0)][string]$class #Class to query
+    )
+    try {
+        $cimObj=Get-CimInstance -class $class
+    } catch {
+        $module.FailJson("There was an error retrieving Windows description $($_.Exception.Message)")
+    }
+    return $cimObj;
+}
+
+#Function to change CIM's object property. In this case used to change computer description
+Function SetWinDescription {
+    param(
+        [Parameter(Position=0)]$object, #CIMs Object to apply the change too
+        [Parameter(Position=0)]$property, #CIMs property to change
+        [Parameter(Position=2)][string]$value #New value for the property
+    )
+    try {
+        Set-CimInstance -InputObject $object -Property @{$property="$value"}
+    } catch {
+        $module.FailJson("There was an error changing Windows description $($_.Exception.Message)")
+    }
+}
+
+#Function to update registry, in order to change license Owner and Organization
+Function WinOwnership {
+    param(
+        [Parameter(Position=0)][ValidateSet("RegisteredOrganization","RegisteredOwner")][string]$type, #Organization or user
+        [Parameter(Position=1)][ValidateSet("set","get")][string]$action, #Get Information or set information
+        [Parameter(Position=2)][String]$value #Value to set
+    )
+	if ($action -eq "get") {
+        try {
+            $regObj=Get-ItemProperty -LiteralPath $regPath
+        } catch {
+            $module.FailJson("There was an error fetching registry $($_.Exception.Message)")
+        }
+		return $regObj.$type
+    }
+    #Upate Owner or Organization
+    if ($action -eq "set") {
+        try {
+            Set-ItemProperty -LiteralPath $regPath -Name $type -Value $value
+        } catch {
+            $module.FailJson("There was an error updating $type $($_.Exception.Message)")
+        }
+    }
+}
+
+#Change description
+if ($description -or $description -eq "") {
+    $descriptionObject=GetCimsObject -class "Win32_OperatingSystem"
+    if ($description -cne $descriptionObject.description) {
+        if (-not $module.CheckMode) {
+            SetWinDescription -object $descriptionObject -property "Description" -value $description
+        }
+        $module.Result.changed = $true
+    }
+}
+
+#Change owner
+if ($owner -or $owner -eq "") {
+    $curentOwner=WinOwnership -type "RegisteredOwner"  -action "get"
+    if ($curentOwner -cne $owner) {
+        if (-not $module.CheckMode) {
+            WinOwnership -type "RegisteredOwner"  -action "set" -value $owner
+        }
+        $module.Result.changed = $true
+    }
+}
+
+#Change organization
+if ($organization -or $organization -eq "") {
+    $curentOrganization=WinOwnership -type "RegisteredOrganization"  -action "get"
```

Comment:
Same thing as above, if you really want you could have `RegisteredOwner` and `RegisteredOrganization` as part of a loop as the check and set operations are the same. The only thing that differs is the property name and the input value, maybe something like

```
$checks = @{
    RegisteredOwner = $owner
    RegisteredOrganization = $organization
}

foreach ($check in $checks.GetEnumerator()) {
    $key = $check.Key
    $value = $check.Value
    if ($null -eq $value) {
        continue
    }

    $actual = (Get-ItemProperty -LiteralPath $regPath -Name $key).$key
    if ($actual -cne $value) {
        Set-ItemProperty -LiteralPath $regPath -Name $key -Value $value -WhatIf:$module.CheckMode
        $module.Result.changed = $true
    }
}
```